client/my-sites/checkout/src/components/restorable-products-context.tsx

```diff
@@ -0,0 +1,28 @@
+import { ResponseCartProduct } from '@automattic/shopping-cart';
+import { createContext, useContext, useState, ReactNode, Dispatch, SetStateAction } from 'react';
+
+interface RestorableProductsContextType {
+	restorableProducts: ResponseCartProduct[];
+	setRestorableProducts: Dispatch< SetStateAction< ResponseCartProduct[] > >;
+}
+
+const RestorableProductsContext = createContext< RestorableProductsContextType | undefined >(
+	undefined
+);
+
+export const RestorableProductsProvider = ( { children }: { children: ReactNode } ) => {
+	const [ restorableProducts, setRestorableProducts ] = useState< ResponseCartProduct[] >( [] );
+	return (
+		<RestorableProductsContext.Provider value={ { restorableProducts, setRestorableProducts } }>
+			{ children }
+		</RestorableProductsContext.Provider>
+	);
+};
```

Comment:
This `value` object will be re-created on every render, forcing the whole tree to re-render. Maybe make the value an array instead.

```suggestion
	const state  = useState< ResponseCartProduct[] >( [] );
	return (
		<RestorableProductsContext.Provider value={ state }>
			{ children }
		</RestorableProductsContext.Provider>
	);
};
```

Reply:
I agree. This is a good idea to prevent re-renders. I believe the array returned from `useState` is stable (I assume?). Although, you'll have to change the hook to expect a tuple instead. Another option would be use to `useMemo()` on the two property object and depend on the state/setter.

Reply:
Yes it is stable.